platform/engineering/design-docs/2020-08-11-redirects-from-cms.md

```diff
@@ -0,0 +1,224 @@
+# Front End Build Redirects from CMS
+
+_Replace the previous line with the the title of your project or component and replace the following lines with your name(s), email and the date._
+**Author(s):** Steve Wirt
+**Last Updated:** Aug 21, 2020
+**Status:** Draft | **In Review** | Approved
+**Approvers:**
+   - [ ] Rian Fowler
+   - [ ] Andrew Gunsch
+   - [ ] Mikki Northuis
+   - [ ] Nick Sullivan
+   - [ ] Taylor Skinner (originally worked on redirect import)
+   - [ ] ??
+
+## Overview
+
+
+### Objective
+The CMS allows editors to create pages and move pages to other locations within the VA.gov website.  The CMS keeps track of when a page's path is changed and builds a redirect from one to the other.  The problem is, there is nothing that moves the redirects known within the CMS to the public website. It is in the best interest of editors, developers and veterans to have the front-end website know about and act on the redirects known to the CMS.
+
+### Background
```

Comment:
can you also include in Background:

* how do redirects currently get to the FE website? I don't think "nothing" is the current answer, my understanding is it's just manual, and this doc should describe that process. (we're wanting to improve the process, rather than to build a process where one doesn't exist, right?)

* description of the redirect review process, i.e. how does IA get involved and review URL redirects today?  (and, in the rest of the doc, how would your proposed accommodate this process?)

Reply:
These are parts that I do not know.  They currently do not get from the CMS to the front end by any automated method.     I can consult with @ncksllvn as I know he has been involved with setting up some of the redirects that currently exits.

Reply:
Current redirect process is documented here: https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/information-architecture/request-redirect.md

Reply:
Does the CMS only automatically build the redirect when the node ID is used?  Also, sort of related to this process, how do we ensure all links across the site are updated appropriately?

Reply:
I don't think we can just automate redirect publishing if it discards the IA and approval processes. Can you speak to where IA approval processes might fit in if we automate this?

I realize this question's a little more product-y than technical since it's more about overall user stories --- but I'm worried this technical proposal changes the user stories in ways that aren't actually OK. curious for your thoughts! (or alternately, is there a PM on your team who's looking at this from the user-stories angle?)

Reply:
+1 to these Qs from Mikki and Andrew about how the governance and oversight process will work. Are there user roles in the CMS that could support that (redirects could be limited to certain user roles)? How else might we preserve a redirect governance model? 

I understand if the governance is not your expertise, Steve! Happy to direct these Qs to the PM for this work, just let me know who I should be talking to.

Reply:
> Does the CMS only automatically build the redirect when the node ID is used? 

@mnorthuis If the path of node/123 gets change from `/here` to `/there` A redirect gets built in the CMS.  If the link in the content is done via the best practice in the CMS then the is updated automatically to show that it is pointing to `/there`.  If however the link is created just in the wysiwyg as an html path then it would still point to `/here`, resulting in a 404 on the front end if we do not have redirects being built with the content releases. 

> Also, sort of related to this process, how do we ensure all links across the site are updated appropriately?

If this proposal were implemented, the link to the redirected page would show up in our node link report that shows on each node for editors.  It would show as a redirected link with a suggestion to fix it.
![image](https://user-images.githubusercontent.com/5752113/91897992-1833f180-ec69-11ea-83ca-6032e2eb28c2.png)